packages/serverless-contracts/src/types/__tests__/constrainedJSONSchema.type.test.ts

```diff
@@ -1,8 +1,50 @@
 import { JSONSchema } from 'json-schema-to-ts';
+import { A } from 'ts-toolbelt';
 
 import { ConstrainedJSONSchema } from '../constrainedJSONSchema';
 
-type Check = ConstrainedJSONSchema extends JSONSchema ? 'pass' : 'fail';
+type ExtendJsonSchemaCheck = A.Extends<ConstrainedJSONSchema, JSONSchema>;
 
-const check: Check = 'pass';
-check;
+const extendJsonSchemaCheck: ExtendJsonSchemaCheck = 1;
+extendJsonSchemaCheck;
+
+const simpleSchema = {
+  type: 'object',
+  properties: { userId: { type: 'string' }, pageNumber: { type: 'string' } },
+  required: ['userId', 'pageNumber'],
+  additionalProperties: false,
+} as const;
+
+type SimpleSchemaCheck = A.Extends<typeof simpleSchema, ConstrainedJSONSchema>;
+
+const simpleSchemaCheck: SimpleSchemaCheck = 1;
+simpleSchemaCheck;
```

Comment:
couldn't we use a generic `Expect<T, V>` or `Assert<T>` to do our type testing instead of this ? 
you can find examples here https://dev.to/tylim88/typescript-test-your-generic-type-part-1-4jbc